nixos/modules/services/databases/postgrest.nix

```diff
@@ -0,0 +1,258 @@
+
+              <https://github.com/PostgREST/postgrest/issues/3956>
+              :::
+            '';
+          };
+
+          server-port = lib.mkOption {
+            type = with lib.types; nullOr port;
+            default = null;
+            example = "3000";
+            description = ''
+              The TCP port to bind the web server.
+            '';
+          };
+
+          server-unix-socket = lib.mkOption {
+            type = with lib.types; nullOr path;
+            default = "/run/postgrest/postgrest.sock";
+            description = ''
+              Unix domain socket where to bind the PostgREST web server.
+            '';
+          };
+        };
+      };
+      default = { };
+      description = ''
+        PostgREST configuration as documented in:
+        <https://docs.postgrest.org/en/stable/references/configuration.html#list-of-parameters>
+
+        ::: {.note}
+        The `settings.jwt-secret` option is blocked.
+        Use `jwtSecretFile` instead.
+        :::
+      '';
+      example = lib.literalExpression ''
+        {
+          db-anon-role = "anon";
+          db-uri = "dbname=postgres";
+          "app.settings.custom" = "value";
+        }
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = (cfg.settings.server-port == null) != (cfg.settings.server-unix-socket == null);
+        message = ''
+          PostgREST can listen either on a TCP port or on a unix socket, but not both.
+          Please set one of `settings.server-port` or `settings.server-unix-socket` to `null`.
+
+          <https://docs.postgrest.org/en/stable/references/configuration.html#server-unix-socket>
+        '';
+      }
+    ];
+
+    warnings =
+      if cfg.settings.admin-server-port != null && cfg.settings.server-host != "127.0.0.1" then
+        [
+          "The PostgREST admin server is potentially listening on a public host. This may expose sensitive information via the `/config` endpoint."
+        ]
+      else
+        [ ];
+
+    systemd.services.postgrest = {
+      description = "PostgREST";
+
+      wantedBy = [ "multi-user.target" ];
+      after = [
+        "network.target"
+        "postgresql.service"
+      ];
+
+      serviceConfig = {
+        DynamicUser = true;
+        LoadCredential =
+          lib.optional (cfg.dbPasswordFile != null) "db_password:${cfg.dbPasswordFile}"
+          ++ lib.optional (cfg.jwtSecretFile != null) "jwt_secret:${cfg.jwtSecretFile}";
+        RuntimeDirectory = "postgrest";
+        User = "postgrest";
+      };
+
+      script =
+        lib.optionalString (cfg.dbPasswordFile != null) ''
+          export PGPASSWORD="$(< "$CREDENTIALS_DIRECTORY/db_password")"
+        ''
+        + lib.optionalString (cfg.jwtSecretFile != null) ''
+          export PGRST_JWT_SECRET="@$CREDENTIALS_DIRECTORY/jwt_secret"
+        ''
+        + ''
+          exec ${lib.getExe pkgs.postgrest} ${configFile}
+        '';
+    };
```

Comment:
Could we somehow use Envirionment instead?

Reply:
Different question: IIRC you're also working on the project itself. Would it be possible to add a PGPASSWORDFILE thingy? IMHO, that's preferable over putting secrets into the process env.

Reply:
I use sops-nix and I can do something like

```
{
  sops.secrets.pgpass = {};
  sops.templates.foobar.pgpassfile = ''
    foo:bar:baz:foobar:${config.sops.placeholder.pgpass}
  '';
}
```

and I'd like to see something like this in nixpgks actually, but that's a different issue. So, personally, I wouldn't mind that actually.